component/main.jsonnet

```diff
@@ -1,10 +1,114 @@
-// main template for floodgate
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
-// The hiera parameters for the component
 local params = inv.parameters.floodgate;
 
-// Define outputs below
+local commonLabels = {
+  'app.kubernetes.io/name': 'floodgate',
+  'app.kubernetes.io/instance': inv.parameters._instance,
+  'app.kubernetes.io/version': params.images.floodgate.tag,
+  'app.kubernetes.io/managed-by': 'commodore',
+  'app.kubernetes.io/component': 'floodgate',
+};
+
+
+local serviceAccount = kube.ServiceAccount('floodgate') {
+  metadata+: {
+    labels+: commonLabels,
+  },
+};
+
+local validateImageDay(d) =
+  assert
+    std.isString(d) :
+    'Floodgate expects parameter `imageDay` to be specified as a string';
+  assert
+    std.length(std.find(d, [ '0', '1', '2', '3', '4', '5', '6' ])) > 0 :
```

Comment:
Why not just map english names ("Sunday" => "0")? This would also solve quoting issues with those string numbers.

Reply:
I considered that, but then you have to decide on at least following design questions: Do you normalize inputs to lower-case, or do you require specific casing? Do you only support fully written out days or also short forms (e.g. "Sun")? If you accept short forms, which ones? Do you still accept day-of-week numbers? If you accept day-of-week numbers, do you automatically quote them if they're provided as integers?

All in all, my impression is that you'd end up with a similar situation to YAML booleans, where magically the value `no` which you intended to be a string gets translated to `false` in your code, if you don't explicitly quote it in YAML.

Reply:
I'd do it the same way Kubernetes does. Case sensitive. Starting with a capital letter. This feels "expected" as I'm used to it in k8s manifests. Everything else is rejected.

I see more advantages for using weekday names: It makes it self-documenting that the parameter is a weekday. It also reduces the mental overhead for non-US users where the first day of the week is not usually Sunday.

From the pod spec:
```yaml
imagePullPolicy: Always
restartPolicy: OnFailure
dnsPolicy: None
```

Reply:
Updated in the latest version